### DISCUSSION TITLE
How to reset `/odom` state?

### DISCUSSION THREAD

**Discussion by alexswerner** [iRobotEducation/create3_docs#363] [Questions] [answered]
Hi,

currently the ROS2 API does seem not expose any ability to reset the odometry published as `/odom`. Is this correct?
I assumed the lifecyle API of the robot_state node could be used for this, but it did not work as expected.
Basically I ran the following commands:
```
ros2 service call /robot_state/change_state lifecycle_msgs/srv/ChangeState "transition:
   id: 4
   label: 'deactivate'"
sleep 10
ros2 service call /robot_state/change_state lifecycle_msgs/srv/ChangeState "transition:
   id: 2
   label: 'cleanup'"
sleep 10
ros2 service call /robot_state/change_state lifecycle_msgs/srv/ChangeState "transition:
   id: 1
   label: 'configure'"
sleep 10
ros2 service call /robot_state/change_state lifecycle_msgs/srv/ChangeState "transition:
   id: 3
   label: 'activate'"
```
And hoped for a position of 0/0/0 to be reported by `/odom`

I'm still using G2.2 here.

**Comment by alsora** [comment, marked as answer]
No, restarting the robot-state node will not affect the odometry.
The reason is that the robot-state node is not responsible for computing the odometry state, but rather it's just reporting a value computed elsewhere.

Note that in version G.3.1 we added a ROS 2 service exactly for this purpose: https://github.com/iRobotEducation/irobot_create_msgs/blob/rolling/srv/ResetPose.srv

```
ros2 service call /reset_pose irobot_create_msgs/srv/ResetPose "{}"
```

This will reset the pose estimate to the origin.

**Reply by alexswerner** [reply]
Thanks a bunch. Also, 10/10 for response time and quality. I'll upgrade asap.
